src/Microsoft.ML.Core/Utilities/DoubleParser.cs

```diff
@@ -70,25 +70,44 @@ public enum Result
             Error = 3
         }
 
-        public static Result Parse(out Single value, string s, int ichMin, int ichLim)
+        /// <summary>
+        /// This produces zero for an empty string.
+        /// </summary>
+        public static bool TryParse(ReadOnlySpan<char> span, out Single value)
+        {
+            var res = Parse(out value, span);
+            Contracts.Assert(res != Result.Empty || value == 0);
+            return res <= Result.Empty;
+        }
+
+        /// <summary>
+        /// This produces zero for an empty string.
+        /// </summary>
+        public static bool TryParse(ReadOnlySpan<char> span, out Double value)
         {
-            Contracts.Assert(0 <= ichMin && ichMin <= ichLim && ichLim <= Utils.Size(s));
+            var res = Parse(out value, span);
+            Contracts.Assert(res != Result.Empty || value == 0);
+            return res <= Result.Empty;
+        }
 
-            for (; ; ichMin++)
+        public static Result Parse(out Single value, ReadOnlySpan<char> span)
```

Comment:
While we are in here fixing this, can we switch the parameters around so the `out` variable is at the end?